src/measure_theory/integral/riesz_markov_kakutani.lean

```diff
@@ -109,3 +112,140 @@ begin
 end
 
 end riesz_subadditive
+
+section urysohn_bounded_nnreal
+
+lemma real.isometry_const_sub (c : ℝ) :
+  isometry (λ (r : ℝ), c - r) :=
+begin
+  intros x y,
+  simp_rw [edist_dist],
+  apply congr_arg ennreal.of_real,
+  simp only [real.dist_eq, sub_sub_sub_cancel_left, abs_sub_comm],
+end
+
+lemma nnreal.lipschitz_with_one_const_sub (c : ℝ≥0) : lipschitz_with 1 (λ (u : ℝ≥0), c - u) :=
+begin
+  have lip₁ : lipschitz_with 1 (coe : ℝ≥0 → ℝ), from lipschitz_with.subtype_coe _,
+  have lip₂ := isometry.lipschitz (real.isometry_const_sub c),
+  have lip₃ : lipschitz_with 1 real.to_nnreal, from lipschitz_with_pos,
+  simpa only [mul_one] using (lip₃.comp lip₂).comp lip₁,
+end
+
+/-- **Urysohn's lemma** for `ℝ≥0`-valued functions:  if `s` and `t` are two disjoint closed sets in
+  a normal topological space `X`, then there exists a continuous function `f : X → ℝ≥0` such that
+
+  * `f` equals zero on `s`;
+  * `f` equals one on `t`;
+  * `f x ≤ 1` for all `x`. -/
+
+lemma exists_bounded_zero_one_of_closed_nnreal {α : Type*} [topological_space α] [normal_space α]
+  {s t : set α} (s_closed : is_closed s) (t_closed : is_closed t) (disj : disjoint s t) :
+  ∃ (f : α →ᵇ ℝ≥0), eq_on f 0 s ∧ eq_on f 1 t ∧ ⇑f ≤ 1 :=
+begin
+  rcases exists_bounded_zero_one_of_closed s_closed t_closed disj
+    with ⟨g, ⟨g_on_s, ⟨g_on_t, g_bdd⟩⟩⟩,
+  use g.nnreal_part,
+  rw bounded_continuous_function.nnreal_part_coe_fun_eq,
+  refine ⟨_, _, _⟩,
+  { intros x hx,
+    simp only [pi.zero_apply, real.to_nnreal_eq_zero, g_on_s hx] },
+  { intros x hx,
+    simp only [comp_app, pi.one_apply, real.to_nnreal_one, g_on_t hx] },
+  { intros x,
+    convert real.to_nnreal_le_to_nnreal (g_bdd x).2,
+    simp only [pi.one_apply, real.to_nnreal_one] },
+end
+
+/--Given two disjoint closed sets `s,t` in a normal space, there exist two bounded continuous
+  `ℝ≥0`-valued functions `f₁,f₂` such that
+
+  * `f₁ = 1` on `s`
+  * `f₂ = 1` on `t`
+  * `f₁ + f₂ = 1`.
+
+-/
+lemma exists_sum_one_of_closed_nnreal {α : Type*} [topological_space α] [normal_space α]
```

Comment:
`sum` suggests it's finitary, but in fact it is binary.
```suggestion
lemma exists_add_eq_one_of_closed_nnreal {α : Type*} [topological_space α] [normal_space α]
```
Feel free to also add the finitary version! It should look like
```
lemma exists_sum_one_of_closed_nnreal {\io α : Type*} [topological_space α] [normal_space α]

```